### ISSUE TITLE
zpool with draid missing/lost all hot spares

### ISSUE THREAD

**Issue by TimLindholm98** [openzfs/zfs#14175]
### System information

Type | Version/2.1.4
 --- | ---
Distribution Name	| Alma Linux
Distribution Version	|  8.6
Kernel Version	| 4.18.0-372.9.1.el8.x86_64
Architecture	| x86_64
OpenZFS Version	| 2.0.7, 2.14

### Describe the problem you're observing
We have a 84 disk SAS attached JBOD through a HBA with 2 paths to each disk.
Aand we created a zpool and with 2 draids, one draid for each jbod shelf (seagate 5U 84 disk JBOD)

draid2:8d:42c:2s-0
draid2:8d:42c:2s-1

The problem is that 2 hot spares from each draid2 is missing from the command "zpool status" in the spares section (see logs). 

But when we first created the pool the spares where visible when we ran "zpool status". Then after the system went into production for about 5 months the system unexpectedly shutdown while in high load.

When the system was booting again it couldnt import the draid pool because the version where missmatched. The kernel loaded zfs/2.0.7 and zfs userspace where 2.1.4. We fixed that with generating a newer initramfs, I dont know why that hadn't happend before automaticly.

So from here we didn't see the hot spares, but we can see them in the zdb output (see logs) and zpool status where the draid is described like this "draid2:8d:42c:2s-0".

They also didnt kick in when 2 disks "FAULTED" probably because the unexpected shutdown

### Describe how to reproduce the problem
We haven't found a way to reproduce the problem we have tried to do virtually but to no success, we did find a way to reproduce the missmatched version but not so the pool lost hot spares. 

### Include any warning/errors/backtraces from the system logs

zpool status -P
```
 pool: hdd0
 state: DEGRADED
status: One or more devices are faulted in response to persistent errors.
	Sufficient replicas exist for the pool to continue functioning in a
	degraded state.
action: Replace the faulted device, or use 'zpool clear' to mark the device
	repaired.
  scan: resilvered 12.6T in 1 days 06:31:15 with 0 errors on Fri Nov 11 00:11:02 2022
config:

	NAME                        STATE     READ WRITE CKSUM
	hdd0                        DEGRADED     0     0     0
	  draid2:8d:42c:2s-0        ONLINE       0     0     0
	    /dev/mapper/upper-d1    ONLINE       0     0     0
	    /dev/mapper/upper-d10   ONLINE       0     0     0
	    /dev/mapper/upper-d11   ONLINE       0     0     0
	    /dev/mapper/upper-d12   ONLINE       0     0     0
	    /dev/mapper/upper-d13   ONLINE       0     0     0
	    /dev/mapper/upper-d14   ONLINE       0     0     0
	    /dev/mapper/upper-d2    ONLINE       0     0     0
	    /dev/mapper/upper-d3    ONLINE       0     0     0
	    /dev/mapper/upper-d4    ONLINE       0     0     0
	    /dev/mapper/upper-d5    ONLINE       0     0     0
	    /dev/mapper/upper-d6    ONLINE       0     0     0
	    /dev/mapper/upper-d7    ONLINE       0     0     0
	    /dev/mapper/upper-d8    ONLINE       0     0     0
	    /dev/mapper/upper-d9    ONLINE       0     0     0
	    /dev/mapper/upper-e1    ONLINE       0     0     0
	    /dev/mapper/upper-e10   ONLINE       0     0     0
	    /dev/mapper/upper-e11   ONLINE       0     0     0
	    /dev/mapper/upper-e12   ONLINE       0     0     0
	    /dev/mapper/upper-e13   ONLINE       0     0     0
	    /dev/mapper/upper-e14   ONLINE       0     0     0
	    /dev/mapper/upper-e2    ONLINE       0     0     0
	    /dev/mapper/upper-e3    ONLINE       0     0     0
	    /dev/mapper/upper-e4    ONLINE       0     0     0
	    /dev/mapper/upper-e5    ONLINE       0     0     0
	    /dev/mapper/upper-e6    ONLINE       0     0     0
	    /dev/mapper/upper-e7    ONLINE       0     0     0
	    /dev/mapper/upper-e8    ONLINE       0     0     0
	    /dev/mapper/upper-e9    ONLINE       0     0     0
	    /dev/mapper/upper-f1    ONLINE       0     0     0
	    /dev/mapper/upper-f10   ONLINE       0     0     0
	    /dev/mapper/upper-f11   ONLINE       0     0     0
	    /dev/mapper/upper-f12   ONLINE       0     0     0
	    /dev/mapper/upper-f13   ONLINE       0     0     0
	    /dev/mapper/upper-f14   ONLINE       0     0     0
	    /dev/mapper/upper-f2    ONLINE       0     0     0
	    /dev/mapper/upper-f3    ONLINE       0     0     0
	    /dev/mapper/upper-f4    ONLINE       0     0     0
	    /dev/mapper/mpathcg     ONLINE       0     0     0
	    /dev/mapper/upper-f6    ONLINE       0     0     0
	    /dev/mapper/upper-f7    ONLINE       0     0     0
	    /dev/mapper/upper-f8    ONLINE       0     0     0
	    /dev/mapper/upper-f9    ONLINE       0     0     0
	  draid2:8d:42c:2s-1        DEGRADED     0     0     0
	    /dev/mapper/bottom-a1   ONLINE       0     0     0
	    /dev/mapper/bottom-a10  ONLINE       0     0     0
	    /dev/mapper/bottom-a11  ONLINE       0     0     0
	    /dev/mapper/bottom-a12  ONLINE       0     0     0
	    /dev/mapper/bottom-a13  ONLINE       0     0     0
	    /dev/mapper/bottom-a14  ONLINE       0     0     0
	    /dev/mapper/bottom-a2   ONLINE       0     0     0
	    /dev/mapper/bottom-a3   ONLINE       0     0     0
	    /dev/mapper/bottom-a4   ONLINE       0     0     0
	    /dev/mapper/bottom-a5   ONLINE       0     0     0
	    /dev/mapper/bottom-a6   ONLINE       0     0     0
	    /dev/mapper/bottom-a7   ONLINE       0     0     0
	    /dev/mapper/bottom-a8   ONLINE       0     0     0
	    /dev/mapper/bottom-a9   ONLINE       0     0     0
	    /dev/mapper/bottom-b1   ONLINE       0     0     0
	    /dev/mapper/bottom-b10  ONLINE       0     0     0
	    /dev/mapper/bottom-b11  ONLINE       0     0     0
	    /dev/mapper/bottom-b12  ONLINE       0     0     0
	    /dev/mapper/bottom-b13  ONLINE       0     0     0
	    /dev/mapper/bottom-b14  ONLINE       0     0     0
	    /dev/mapper/bottom-b2   ONLINE       0     0     0
	    /dev/mapper/bottom-b3   ONLINE       0     0     0
	    /dev/mapper/bottom-b4   ONLINE       0     0     0
	    /dev/mapper/bottom-b5   ONLINE       0     0     0
	    /dev/mapper/bottom-b6   FAULTED    186   611     0  too many errors
	    /dev/mapper/bottom-b7   ONLINE       0     0     0
	    /dev/mapper/bottom-b8   ONLINE       0     0     0
	    /dev/mapper/bottom-b9   ONLINE       0     0     0
	    /dev/mapper/bottom-c1   ONLINE       0     0     0
	    /dev/mapper/bottom-c10  ONLINE       0     0     0
	    /dev/mapper/bottom-c11  ONLINE       0     0     0
	    /dev/mapper/bottom-c12  ONLINE       0     0     0
	    /dev/mapper/bottom-c13  ONLINE       0     0     0
	    /dev/mapper/bottom-c14  ONLINE       0     0     0
	    /dev/mapper/bottom-c2   ONLINE       0     0     0
	    /dev/mapper/bottom-c3   ONLINE       0     0     0
	    /dev/mapper/bottom-c4   ONLINE       0     0     0
	    /dev/mapper/bottom-c5   ONLINE       0     0     0
	    /dev/mapper/bottom-c6   ONLINE       0     0     0
	    /dev/mapper/bottom-c7   ONLINE       0     0     0
	    /dev/mapper/bottom-c8   ONLINE       0     0     0
	    /dev/mapper/bottom-c9   ONLINE       0     0     0

errors: No known data errors
```

[zdb-C.txt](https://github.com/openzfs/zfs/files/9990518/zdb-C.txt)
[zfs-get-all-hdd0.txt](https://github.com/openzfs/zfs/files/9990532/zfs-get-all-hdd0.txt)
[zpool-event-verbose.txt](https://github.com/openzfs/zfs/files/9990533/zpool-event-verbose.txt)
[zpool-event.txt](https://github.com/openzfs/zfs/files/9990534/zpool-event.txt)
[zpool-guid.txt](https://github.com/openzfs/zfs/files/9990535/zpool-guid.txt)
[zpool-list-verbose.txt](https://github.com/openzfs/zfs/files/9990536/zpool-list-verbose.txt)
[zpool-path.txt](https://github.com/openzfs/zfs/files/9990537/zpool-path.txt)
[zpool-verbose.txt](https://github.com/openzfs/zfs/files/9990538/zpool-verbose.txt)

**Comment by ryao** [comment]
This is a shot in the dark, but have you tried 2.1.6? I belief that some draid fixes have been done since 2.1.4, although I am not sure offhand if they apply here.

**Comment by TimLindholm98** [comment]
> This is a shot in the dark, but have you tried 2.1.6? I believe that some draid fixes have been done since 2.1.4, although I am not sure offhand if they apply here.

Its a production system with no backup of the pool data, so updating the system feels a bit risky. We will get another identical server so we can back it up then do som more "risky" actions.

Updating the ZFS is a option i have considered and I think it could fix it but I havent read anything that immediately caught my attention.
